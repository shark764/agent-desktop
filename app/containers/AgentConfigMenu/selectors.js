import { createSelector } from 'reselect';

/**
* Other specific selectors
*/

const selectToolbarDomain = (state) => state.get('toolbar');

const selectEnabledStats = createSelector(
  selectToolbarDomain,
  (toolbar) => toolbar.get('enabledStats').toJS()
);

const selectAvailableStats = createSelector(
  selectToolbarDomain,
  (toolbar) => toolbar.get('availableStats').toJS()
);

export {
  selectEnabledStats,
  selectAvailableStats,
};
