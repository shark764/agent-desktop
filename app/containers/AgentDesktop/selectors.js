/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import { createSelector } from 'reselect';

const selectLoginMap = (state) => state.get('login');
const selectAgentDesktopMap = (state) => state.get('agentDesktop');

const selectAgentId = createSelector(
  selectLoginMap,
  (login) => login.get('agent').get('userId')
);

const selectIsAgentReady = createSelector(
  selectAgentDesktopMap,
  (agentDesktop) => agentDesktop.get('presence') === 'ready'
);

const selectInteractions = createSelector(
  selectAgentDesktopMap,
  (agentDesktop) => agentDesktop.get('interactions')
);

const getSelectedInteractionId = createSelector(
  selectAgentDesktopMap,
  (agentDesktop) => agentDesktop.get('selectedInteractionId')
);

const selectNewInteractionPanel = createSelector(
  selectAgentDesktopMap,
  (agentDesktop) => agentDesktop.get('newInteractionPanel')
);
const selectQueues = createSelector(
  selectAgentDesktopMap,
  (agentDesktop) => agentDesktop.get('queues').toJS()
);

const selectSelectedInteraction = createSelector(
  [selectNewInteractionPanel, selectInteractions, getSelectedInteractionId],
  (newInteractionPanel, interactions, selectedInteractionId) => {
    if (selectedInteractionId !== undefined) {
      if (selectedInteractionId === 'creating-new-interaction') {
        return newInteractionPanel.toJS();
      } else {
        return interactions.toJS().find((interaction) =>
          interaction.interactionId === selectedInteractionId
        );
      }
    } else {
      return undefined;
    }
  }
 );

const selectAwaitingDisposition = createSelector(
  selectSelectedInteraction,
  (interaction) => (
    typeof interaction !== 'undefined' &&
    interaction.status === 'wrapup' &&
    interaction.dispositionDetails.forceSelect &&
    interaction.dispositionDetails.selected.length === 0
  )
);

const selectHasVoiceInteraction = createSelector(
  selectInteractions,
  (interactions) => interactions.findIndex(
    (interaction) => (interaction.get('channelType') === 'voice')
  ) !== -1
);

const selectSmsInteractionNumbers = createSelector(
  selectInteractions,
  (interactions) => {
    const smsInteractionNumbers = [];
    interactions.forEach((interaction) => {
      if (interaction.get('channelType') === 'sms') {
        smsInteractionNumbers.push(interaction.get('customer'));
      }
    });
    return smsInteractionNumbers;
  }
);

const selectIsContactsPanelCollapsed = createSelector(
  selectAgentDesktopMap,
  (agentDesktop) => agentDesktop.get('isContactsPanelCollapsed')
);

export {
  selectAgentId,
  selectIsAgentReady,
  selectInteractions,
  getSelectedInteractionId,
  selectLoginMap,
  selectAgentDesktopMap,
  selectSelectedInteraction,
  selectAwaitingDisposition,
  selectHasVoiceInteraction,
  selectSmsInteractionNumbers,
  selectQueues,
  selectIsContactsPanelCollapsed,
};
