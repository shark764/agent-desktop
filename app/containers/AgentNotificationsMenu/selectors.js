/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

const selectNotificationPreferences = (state) =>
  state.get('notificationPreferences');

export const selectAudioPreferences = (state) =>
  selectNotificationPreferences(state).get('audioNotifications');

export const selectVisualPreferences = (state) =>
  Notification.permission === 'granted' &&
  selectNotificationPreferences(state).get('visualNotifications');
