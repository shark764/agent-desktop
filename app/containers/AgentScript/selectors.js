import { createSelector } from 'reselect';

/**
 * Direct selector to the agentScript state domain
 */
const selectAgentScriptDomain = () => (state) => state.get('agentScript');

/**
 * Other specific selectors
 */


/**
 * Default selector used by AgentScript
 */

const selectAgentScript = () => createSelector(
  selectAgentScriptDomain(),
  (substate) => substate.toJS()
);

export default selectAgentScript;
export {
  selectAgentScriptDomain,
};
