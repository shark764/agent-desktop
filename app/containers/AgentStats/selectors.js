import { createSelector } from 'reselect';

/**
* Other specific selectors
*/

const selectToolbarDomain = (state) => state.get('toolbar');
const selectLoginDomain = (state) => state.get('login');

const selectEnabledStats = createSelector(
  selectToolbarDomain,
  (toolbar) => toolbar.get('enabledStats').toJS()
);

const selectAvailableStats = createSelector(
  selectToolbarDomain,
  (toolbar) => toolbar.get('availableStats').toJS()
);

const selectCurrentAgent = createSelector(
  selectLoginDomain,
  (login) => login.get('agent').toJS()
);

export {
  selectEnabledStats,
  selectAvailableStats,
  selectCurrentAgent,
};
