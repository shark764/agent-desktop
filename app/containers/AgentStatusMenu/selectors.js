import { createSelector } from 'reselect';

const selectAgentDesktopDomain = (state) => state.get('agentDesktop');

const selectHasActiveInteractions = createSelector(
  selectAgentDesktopDomain,
  (agentDesktop) =>
    agentDesktop.get('interactions').find((interaction) =>
      interaction.get('status') === 'work-accepting' || interaction.get('status') === 'work-accepted'
    ) !== undefined
);

const selectExtensions = createSelector(
  selectAgentDesktopDomain,
  (agentDesktop) =>
    agentDesktop.get('extensions').toJS()
);

const selectActiveExtension = createSelector(
  selectAgentDesktopDomain,
  (agentDesktop) =>
    agentDesktop.get('activeExtension').toJS()
);

const selectHasActiveWrapup = createSelector(
  selectAgentDesktopDomain,
  (agentDesktop) =>
    agentDesktop.get('interactions').find((interaction) =>
      interaction.get('status') === 'wrapup'
    ) !== undefined
);

export {
  selectHasActiveInteractions,
  selectExtensions,
  selectActiveExtension,
  selectHasActiveWrapup,
};
