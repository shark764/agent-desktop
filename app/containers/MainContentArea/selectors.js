/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import { createSelector } from 'reselect';

const selectAgentDesktopDomain = (state) => state.get('agentDesktop');

const selectMessageTemplates = createSelector(
  selectAgentDesktopDomain,
  (agentDesktop) => agentDesktop.get('userConfig').get('messageTemplates').toJS()
);

export {
  selectMessageTemplates,
};
