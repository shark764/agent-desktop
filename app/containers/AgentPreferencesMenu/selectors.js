/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */

import { createSelector } from 'reselect';

import { getCurrentTenantPermissions } from 'containers/App/selectors';

export const selectHasViewStatsPermission = createSelector(
  [getCurrentTenantPermissions],
  (tenantPermissions) =>
    tenantPermissions.includes('AGENT_EXPERIENCE_METRICS_VIEW')
);
