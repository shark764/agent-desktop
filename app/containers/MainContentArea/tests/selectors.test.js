import { fromJS } from 'immutable';
import { selectMessageTemplates } from '../selectors';

describe('selectMessageTemplates', () => {
  it('should filter out active messaging templates from those assigned to agents', () => {
    expect(
      selectMessageTemplates(
        fromJS({
          agentDesktop: {
            userConfig: {
              messageTemplates: [{ active: true }, { active: false }],
            },
          },
        })
      )
    ).toMatchSnapshot();
  });
  it('should return empty array when no messaging-templates are assigned to agents', () => {
    expect(
      selectMessageTemplates(
        fromJS({ agentDesktop: { userConfig: { messageTemplates: [] } } })
      )
    ).toEqual([]);
  });
});
