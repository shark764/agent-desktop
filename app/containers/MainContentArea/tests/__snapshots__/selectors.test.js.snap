// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectMessageTemplates should filter out active messaging templates from those assigned to agents 1`] = `
Array [
  Object {
    "active": true,
  },
]
`;
